include/swift/AST/Decl.h

```diff
@@ -6060,6 +6074,12 @@ class AbstractFunctionDecl : public GenericContext, public ValueDecl {
   /// constructor.
   bool hasDynamicSelfResult() const;
 
+  /// Get all derivative function configurations.
+  ArrayRef<AutoDiffConfig> getDerivativeFunctionConfigurations();
```

Comment:
Note: `AbstractFunctionDecl::getDerivativeFunctionConfigurations` is used for derivative function lookup (SILOptimizer stuff) and actually isn't relevant for `@differentiable` attribute type-checking. For now, `ASTContext::DifferentiableAttrs` is used to check duplicate `@differentiable` attributes.

I realized this too late - TF-1100 tracks the rest of upstreaming `AbstractFunctionDecl::getDerivativeFunctionConfigurations`.

Reply:
Actually, `AbstractFunctionDecl::getDerivativeFunctionConfigurations` is used in `TypeCheckProtocol.cpp` for verifying that protocol requirement `@differentiable` attributes are satisfied by witnesses.